### PR TITLE
Telemetry Event Payload Validation

### PR DESCRIPTION
**Pull request by mehulsonowal**
## Description

Validating required fields for telemetry messages.

## Check list

Your PR is not ready to be reviewed? Please save it as a draft :pray:

- [ ] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [ ] CI is passing
- [ ] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

File: parametric/test_trace_distributed.py [hunk 1]
```diff
@@ -62,8 +62,7 @@ def test_distributed_headers_extract_datadog_invalid_D002(test_agent, test_libra
 @pytest.mark.skip_library("golang", "not implemented")
 @pytest.mark.skip_library("nodejs", "not implemented")
 def test_distributed_headers_inject_datadog_D003(test_agent, test_library):
-    """Ensure that Datadog distributed tracing headers are injected properly.
-    """
+    """Ensure that Datadog distributed tracing headers are injected properly."""
```
Comment (Kyle-Verhoog) [inline parametric/test_trace_distributed.py:65]: can we do the formatting changes separately? It makes this PR harder to review and introduces irrelevant changes

Comment (mehulsonowal) [reply parametric/test_trace_distributed.py]: yeah, sorry about that! something went wrong with black, but i reverted the changes

**Review by cbeauchesne — COMMENTED** [review]
Suggestion : use json schema in `utils/interfaces/schemas` to validate schema-related assertion.

**Comment by cbeauchesne** [comment]
There are some conflict. and some weblog are not passing. If you want to add an exception (an error in a tracer on a given point of the schema), please let me know.

**Review by cbeauchesne — APPROVED** [review]

File: utils/interfaces/schemas/library/telemetry/proxy/api/v2/apmtelemetry-request.json [hunk 2]
```diff
@@ -605,7 +605,7 @@
 					"$ref": "#/$defs/Error"
 				}
 			},
-			"required": ["enabled"]
+			"$comment": "enabled should be a required field, but currently not implemented by python"
```
Comment (cbeauchesne) [inline utils/interfaces/schemas/library/telemetry/proxy/api/v2/apmtelemetry-request.json:608]: You have a mechanism that allow ignoring some errors under condition, it's better than having a wrong schema for everybody. Feel free to add a meeting, we can do it together.

**Comment by cbeauchesne** [comment]
@mehulsonowal 

About [cafacd7](https://github.com/DataDog/system-tests/pull/617/commits/cafacd7c1024d4d4bba643469ff0343cb5ec5483), configuration seems to be an array, can you confirm ? 

And side note, on .NET, value can be missing, is it an error ?

**Review by cbeauchesne — APPROVED** [review]

**Review by ameske — APPROVED** [review]
👍🏻 for remote config, but I have a question about why this change required changes in the RC scenario. @cbeauchesne - it looks like the changes to the remote config scenario are there because we're trying to run a full schema check as well. Seems like we shouldn't want to care about these non-rc related schema errors in our scenario. Should we clean this up so we can remove it?

**Comment by cbeauchesne** [comment]
@ameske , during remote config scenario, if i'm not wrong, you may want to validate data sent by the tracer using schema (see `utils/interfaces/schemas/library/v0.7/config-request.json` ? ), data that are not sent in the default scenario.

**Pull request merged**